rusoto/credential/src/lib.rs

```diff
@@ -100,9 +101,14 @@ impl AwsCredentials {
 
     /// Determine whether or not the credentials are expired.
     fn credentials_are_expired(&self) -> bool {
-        // This is a rough hack to hopefully avoid someone requesting creds then sitting on them
-        // before issuing the request:
-        self.expires_at < Utc::now() + Duration::seconds(20)
+        match self.expires_at {
+            Some(ref e) =>
+                // This is a rough hack to hopefully avoid someone requesting creds then sitting on them
```

Comment:
This really looks like an ugly hack, shouldn't we remove the additional 20 seconds. Client must always expect that their credentials expired or have been revoked anyway, so they must be prepared to handle such errors anyway.

Reply:
I believe the use case is direct use of the instance profile provider: it's possible to get credentials and not use them for a while, where they could expire. We also want to preemptively fetch the new credentials when they are available. So if ours are due to expire within 20 seconds, we fetch new ones to avoid handing back credentials that could be expired.

Is it the right thing to do? I'm not sold it's terrible, but we could probably do better. 😄